scripts/update-browser-releases/edge.ts

```diff
@@ -0,0 +1,197 @@
+
+import * as fs from 'node:fs';
+
+import chalk from 'chalk-template';
+
+import { newBrowserEntry, updateBrowserEntry } from './utils.js';
+
+/**
+ * getReleaseNotesURL - Guess the URL of the release notes
+ * @param {string} status The status of the release
+ * @param {string} fullVersion The version of the release
+ * @param {string} date The date of the release
+ * @returns {string} The URL of the release notes or the empty string if not found
+ */
+const getReleaseNotesURL = async (status, fullVersion, date) => {
+  // If the status isn't stable, do not give back any release notes.
+  if (status !== 'Stable') {
+    return '';
+  }
+  console.log(fullVersion);
+  const versionStr = fullVersion.replace(/\./g, '');
+  const month = [
+    'january',
+    'february',
+    'march',
+    'april',
+    'may',
+    'june',
+    'july',
+    'august',
+    'september',
+    'october',
+    'november',
+    'december',
+  ];
+  const dateObj = new Date(date);
+  const dateStr = `${
+    month[dateObj.getMonth()]
+  }-${dateObj.getDate()}-${dateObj.getFullYear()}`;
+  return `https://learn.microsoft.com/en-us/deployedge/microsoft-edge-relnote-stable-channel#version-${versionStr}-${dateStr}`;
+};
+
+/**
+ * updateEdgeReleases - Update the json file listing the browser releases of an Edge browser
+ * @param {object} options The list of options for Edge.
+ */
+export const updateEdgeReleases = async (options) => {
+  //
+  // Get the JSON with the versions from chromestatus
+  //
+  const buffer = await fetch(options.edgeupdatesURL);
+  const edgeVersions = JSON.parse(await buffer.text());
+
+  //
+  // Get the chrome.json from the local BCD
+  //
+  const file = fs.readFileSync(`${options.bcdFile}`);
+  const edgeBCD = JSON.parse(file.toString());
+
+  //
+  // Update the three channels
+  //
+  const channels = new Map([
+    ['current', options.releaseBranch],
+    ['beta', options.betaBranch],
+    ['nightly', options.nightlyBranch],
+  ]);
+
+  // Information store to be used outside the for loop
+  const data = {};
+
+  for (const [key, value] of channels) {
+    //
+    // Extract the useful data
+    //
+
+    // Find the right entry
+    const entry = edgeVersions.find((e) => e['Product'] === value);
+
+    // Get the last version
+    const version = parseFloat(entry['Releases'][0]['ProductVersion']);
+
+    // We need to find the first release with this version:
+    // The full version number is needed for the release note URL
+    // as well as the date
+    let id = 1;
+    while (
+      id < entry['Releases'].length &&
+      parseFloat(entry['Releases'][id]['ProductVersion']) === version
+    ) {
+      id++;
+    }
+    id--;
+
+    data[value] = {};
+    data[value].version = parseFloat(entry['Releases'][id]['ProductVersion']);
+    data[value].fullVersion = entry['Releases'][id]['ProductVersion'];
+
+    // Get published date
+    data[value].versionDate = entry['Releases'][id]['PublishedTime'].substring(
```

Comment:
Yeah, you're right, the `PublishedTime` isn't ideal unfortunately. I didn't realize this when looking at the date.
I will try to get us to add an `ExpectedReleaseDate` (or something like this) for Dev and Beta. No promises, but that would be much better.
In the meantime, would it make sense to continue manually maintaining the Dev and Beta dates?

Reply:
@captainbrosset 
Yes, we can skip this one now; it must be added manually. (We can even add a reminder to double-check this date in the PR that will be created from the change.)

We can update the script when we have it in the API.

I will ask for another review when this is done.

Reply:
Solved by accessing the MD file with the release notes.